### ISSUE TITLE
0.17.1 runs at 100% CPU till Firefox asks to stop it

### ISSUE THREAD

**Issue by cryptomilk** [chzesa/tiled-tab-groups#4]
With Firefox 65.0.1 and 0.17.1 when I start Tiled Tab Groups it runs at 100% CPU for a while till Firefox asks to stop the script.

Is there a way how I find out what is going on and provide more information?

Thanks for the great module.

**Comment by chzesa** [comment]
Typing `about:debugging` in the url bar brings up the addon debugging page. Click debug under tiled tab groups and copy the `Error: Script terminated by timeout at:` message. If debug is greyed out, enable debugging at the top of the page.

**Comment by cryptomilk** [comment]
```
Loading failed for the <script> with source “moz-extension://b8018c41-cd3e-4f00-9c6c-270701374959/scripts/plural-data.js”. brief.xhtml:15:4
The character encoding of the HTML document was not declared. The document will render with garbled text in some browser configurations if the document contains characters from outside the US-ASCII range. The character encoding of the page must be declared in the document or in the transfer protocol. extension.html
The ‘storage’ attribute in options passed to indexedDB.open is deprecated and will soon be removed. To get persistent storage, please use navigator.storage.persist() instead. database.js:145:21
ReferenceError: "groupId is not defined"

    groupInterface moz-extension://17b0f160-af55-4af1-9857-3cd4a5ba9cb3/js/groupInterface.js:152 

src.js:16:5
TypeError: "TABINTERFACE is undefined"

    init moz-extension://17b0f160-af55-4af1-9857-3cd4a5ba9cb3/js/background.js:429 guard moz-extension://17b0f160-af55-4af1-9857-3cd4a5ba9cb3/external/js-syncqueue/src.js:13 guard moz-extension://17b0f160-af55-4af1-9857-3cd4a5ba9cb3/external/js-syncqueue/src.js:11 execute moz-extension://17b0f160-af55-4af1-9857-3cd4a5ba9cb3/external/js-syncqueue/src.js:34 do moz-extension://17b0f160-af55-4af1-9857-3cd4a5ba9cb3/external/js-syncqueue/src.js:59 init moz-extension://17b0f160-af55-4af1-9857-3cd4a5ba9cb3/js/background.js:426 
```

When I stop the script all I see is:

```
Error: Script terminated by timeout at:
```

It doesn't provide a location in the source code :-(

**Comment by cryptomilk** [comment]
I've tried a view different ways and finally got it:

```
Error: Script terminated by timeout at:
guard@moz-extension://17b0f160-af55-4af1-9857-3cd4a5ba9cb3/external/js-syncqueue/src.js:11:10
execute@moz-extension://17b0f160-af55-4af1-9857-3cd4a5ba9cb3/external/js-syncqueue/src.js:34:10
src.js:11:10
```

**Comment by chzesa** [comment]
Should be fixed with 2e3265b328ab4f628fb173023b2390aa8018b622
I've disabled versions 0.17-0.17.2, new version is up on AMO and you should be able to update.

**Comment by cryptomilk** [comment]
I can confirm that it is fixed with 0.17.3! Thank you very much for the fast response and fix!
